### PR TITLE
Add a generic Set and SetDefault to SettingsHolder

### LINKED ISSUE #5182: Add a generic SettingsHolder.Set
so this 
https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/Settings/SettingsHolder.cs#L204

```
public void Set<T>(object value)
```

It would be nice to have a 

```
public void Set<T>(T value)
```

This way instead of doing this

```
var settings = new SettingsHolder();
var diagnosticEntries = new StartupDiagnosticEntries();
settings.Set<StartupDiagnosticEntries>(diagnosticEntries);
```

I could do this 

```
var settings = new SettingsHolder();
var diagnosticEntries = new StartupDiagnosticEntries();
settings.Set(diagnosticEntries);
```

And the same for `SetDefault` 
https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/Settings/SettingsHolder.cs#L224

### PR DESCRIPTION
**Pull request by SeanFeldman**
Fixes #5182 by adding generic overloads

### CODE REVIEW

File: src/NServiceBus.Core/Settings/SettingsHolder.cs [hunk 1]
```diff
@@ -206,6 +206,16 @@ public void Set<T>(object value)
             Set(typeof(T).FullName, value);
         }
 
+        /// <summary>
+        /// Sets the value .
```
Comment (bording) [inline src/NServiceBus.Core/Settings/SettingsHolder.cs]: :fire: the space after "value"

Comment (bording) [inline src/NServiceBus.Core/Settings/SettingsHolder.cs]: :fire: the space after "value"

**Comment by bording** [comment]
I'm also thinking that maybe we should consider marking the `public void Set<T>(object value)` and `public void SetDefault<T>(object value)` overloads as obsolete now? With these new overloads, there's no reason to keep the `object` versions that I can see.

**Comment by SeanFeldman** [comment]
Obsolete with warning, right @bording ?

**Comment by bording** [comment]
> Obsolete with warning, right @bording ?

Yes, that's what I was thinking. However, before you do that, let's see what other @Particular/nservicebus-maintainers think of the idea.

**Comment by timbussmann** [comment]
> I'm also thinking that maybe we should consider marking the public void Set<T>(object value) and public void SetDefault<T>(object value) overloads as obsolete now? 

sounds reasonable to me. As long as we have the `Set(key, value)` overload, there should be no risk of potentially breaking any usage patterns.

**Comment by SeanFeldman** [comment]
Discussed with @timbussmann and we've decided to leave obsoletion out of this PR, submitting it as a separate PR.

@bording could you please update your review? Thank you.

**Pull request merged**